### ISSUE TITLE
Weird "'T' is declared but never used" error

### ISSUE THREAD

**Issue by vilic** [palantir/tslint#2451]
### Bug Report

- __TSLint version__: 5.0.0
- __TypeScript version__: 2.2.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

I cannot find a minimal reproducing snippet so far.

Please try

```sh
git clone https://github.com/vilic/clime.git
cd clime
npm install
npm run lint -- --type-check
```

#### Actual behavior

ERROR: C:/Projects/clime/src/castable/array.ts[12, 38]: 'T' is declared but never used.

#### Expected behavior

No error.

**Comment by andy-ms** [comment]
May be related to Microsoft/TypeScript#14953. But I can't reproduce this using your steps.

**Comment by vilic** [comment]
@andy-ms Okay, thanks for the workaround.

**Comment by andy-ms** [comment]
This is probably a duplicate of #2470.

**Comment by morozp** [comment]
For my mind it is bad decision. (remove role from lineter)
I think that 'ts-nameof-loader'  should remove imports for types that used only for ts-nameof
